### ISSUE TITLE
Verify Pagination Configuration

### ISSUE THREAD

**Issue by bbernays** [cloudquery/cloudquery#4020]
All paginated requests should have MaxResults set to the maximum for that specific API to reduce total number of API calls

**Comment by disq** [comment]
I've looked at this before, most of them don't document what the allowed `MaxResults` is, and some just don't do any pagination (and tend to get all resources) if you don't specify any (instead of defaulting to a sane number of items per page).

**Comment by shimonp21** [comment]
`Two` google-sheets showing the impact of the change from default `maxResults` to 100 in `aws_servicequotas_services`.

https://docs.google.com/spreadsheets/d/1X8j8fN5cOqH5Bdx2ycHlv__L7QWwo0EZGwQHF1hrjPk/edit?usp=sharing
https://docs.google.com/spreadsheets/d/15VXZ8hzU58L_varzKqQks0MBTkFQHmNBZFnxUA6DPFA/edit?usp=sharing

You can see that with default config, the max is apparently 20, and required 369 API calls in the largest region `aws_servicequotas_quotas`.
While with `maxResults=100`, required only 249.

**Comment by bbernays** [comment]
Great work on that, but wasn't that work part of this issue: #3728

**Comment by shimonp21** [comment]
Looked a little more at the MaxResults feature today. 
My feeling is that it is not possible to (safely) do automatically.
Changing MaxResults to a too-high value will cause API calls to fail.
However, the documentation is too spotty (and sometimes WRONG) to confidently release these changes without testing.
[e.g. https://docs.aws.amazon.com/appstream2/latest/APIReference/API_DescribeUserStackAssociations.html specifies MaxResults=500 , but calling with 500 returns an error claiming the actual allowed value is 25 : "operation error AppStream: DescribeUserStackAssociations, https response error StatusCode: 400, RequestID: 2d774eea-65a1-48e6-be61-9e1ecc852bda, api error InvalidParameterValueException: limit must be less than 25"

However, because many tables are nested, they are impossible to test without creating parent-resources in AWS.
However - we don’t have an integration environment with example resources, so this test will require a lot of manual work PER-RESOURCE.
In conclusion I think we should skip this feature for now (but we should make use of maxResults for new resources!)

**Comment by shimonp21** [comment]
Closing as a 'Won't fix' for now.
